postgres/client.js

```diff
@@ -102,21 +180,112 @@ function batch(ops) {
  * @return {Stream}
  */
 function createReadStream(options) {
-  const queryStream = new QueryStream(READ_STREAM(options)),
-    transform = TransformStream(options);
+  const { prefix, values, keys } = options,
+    { schema, table } = findSchemaAndTable(prefix),
+    transform = TransformStream(options),
+    selects = [];
+  var stream;
+
+  if (keys) selects.push('id');
+  if (values) selects.push('data');
 
-  client.query(queryStream).pipe(transform);
+  if (schema) {
+    stream = knex.select(...selects).withSchema(schema).from(table).where('id', 'like', `${prefix}%`).stream();
+  } else {
+    stream = knex.select(...selects).from(table).where('id', 'like', `${prefix}%`).stream();
+  }
+
+  stream.pipe(transform);
 
   return transform;
 }
 
+/**
+ * [putMeta description]
+ * @param  {[type]} key   [description]
+ * @param  {[type]} value [description]
+ * @return {[type]}       [description]
+ */
+function putMeta(key, value) {
+  const { schema, table } = findSchemaAndTable(key);
+
+  return onConflictPut(key, parseOrNot(value), 'meta', schema, table);
+}
+
+/**
+ * [putMeta description]
+ * @param  {[type]} key   [description]
+ * @param  {[type]} value [description]
+ * @return {[type]}       [description]
+ */
+function getMeta(key) {
+  const { schema, table } = findSchemaAndTable(key);
+
+  if (schema) {
+    return knex.withSchema(schema)
+      .select('meta')
+      .from(table)
+      .where('id', key)
+      .get('rows')
+      .then(resp => resp[0].meta);
+  }
+
+  return knex
+    .select('meta')
+    .from(table)
+    .where('id', key)
+    .then(resp => resp[0].meta);
+}
+
+/**
+ * Creates a table with the name that's
+ * passed into the function. Table has
+ * an `id` (text) and `data` (jsonb) columns
+ *
+ * @param {String} table
+ * @returns {Promise}
+ */
+function createTable(table) {
+  return knex.raw(`CREATE TABLE IF NOT EXISTS ${table} ( id TEXT PRIMARY KEY NOT NULL, data JSONB );`);
```

Comment:
You can use bindings across all your raw invokations for a bit of added safety:

```js
return knex.raw('CREATE TABLE IF NOT EXISTS ?? ( id TEXT PRIMARY KEY NOT NULL, data JSONB, meta JSONB );', [table]);
```

Note: using the `??` to notify it as an identifier and not as a value